From 8205b66a1104171284699985409ddd4d6921400d Mon Sep 17 00:00:00 2001
From: Andrew Waterman <waterman@cs.berkeley.edu>
Date: Fri, 26 Aug 2016 19:59:08 -0700
Subject: [PATCH 07/10] Use user-mode wall-clock timer

The most recent ISA spec no longer has privileged timers.

Reverts 7cf4f80bf6fdcb518ae98caead790635b687b549
---
 arch/riscv/include/asm/timex.h | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/arch/riscv/include/asm/timex.h b/arch/riscv/include/asm/timex.h
index 9aaff9c..45f2e36 100644
--- a/arch/riscv/include/asm/timex.h
+++ b/arch/riscv/include/asm/timex.h
@@ -12,16 +12,16 @@ static inline cycles_t get_cycles(void)
 #ifdef __riscv64
 	cycles_t n;
 	__asm__ __volatile__ (
-		"csrr %0, stime"
+		"rdtime %0"
 		: "=r" (n));
 	return n;
 #else
 	u32 lo, hi, tmp;
 	__asm__ __volatile__ (
 		"1:\n"
-		"csrr %0, stimeh\n"
-		"csrr %1, stime\n"
-		"csrr %2, stimeh\n"
+		"rdtimeh %0\n"
+		"rdtime %1\n"
+		"rdtimeh %2\n"
 		"bne %0, %2, 1b"
 		: "=&r" (hi), "=&r" (lo), "=&r" (tmp));
 	return ((u64)hi << 32) | lo;
-- 
2.10.2

